src/coord/src/catalog/builtin.rs

```
@@ -0,0 +1,465 @@
+JOIN mz_catalog.mz_catalog_names mcn_source ON mcn_source.global_id = frontier_source.global_id",
+    id: GlobalId::System(3011),
+};
+
+lazy_static! {
+    pub static ref BUILTINS: BTreeMap<GlobalId, Builtin> = {
+        let builtins = vec![
+            Builtin::Log(&MZ_DATAFLOW_OPERATORS),
+            Builtin::Log(&MZ_DATAFLOW_OPERATORS_ADDRESSES),
+            Builtin::Log(&MZ_DATAFLOW_CHANNELS),
+            Builtin::Log(&MZ_SCHEDULING_ELAPSED),
+            Builtin::Log(&MZ_SCHEDULING_HISTOGRAM),
+            Builtin::Log(&MZ_SCHEDULING_PARKS),
+            Builtin::Log(&MZ_ARRANGEMENT_SIZES),
+            Builtin::Log(&MZ_ARRANGEMENT_SHARING),
+            Builtin::Log(&MZ_MATERIALIZATIONS),
+            Builtin::Log(&MZ_MATERIALIZATION_DEPENDENCIES),
+            Builtin::Log(&MZ_WORKER_MATERIALIZATION_FRONTIERS),
+            Builtin::Log(&MZ_PEEK_ACTIVE),
+            Builtin::Log(&MZ_PEEK_DURATIONS),
+            Builtin::Table(&MZ_VIEW_KEYS),
+            Builtin::Table(&MZ_VIEW_FOREIGN_KEYS),
+            Builtin::Table(&MZ_CATALOG_NAMES),
+            Builtin::Table(&MZ_KAFKA_SINKS),
+            Builtin::Table(&MZ_AVRO_OCF_SINKS),
+            Builtin::View(&MZ_ADDRESSES_WITH_UNIT_LENGTHS),
+            Builtin::View(&MZ_DATAFLOW_NAMES),
+            Builtin::View(&MZ_DATAFLOW_OPERATOR_DATAFLOWS),
+            Builtin::View(&MZ_RECORDS_PER_DATAFLOW_OPERATOR),
+            Builtin::View(&MZ_RECORDS_PER_DATAFLOW),
+            Builtin::View(&MZ_RECORDS_PER_DATAFLOW_GLOBAL),
+            Builtin::View(&MZ_PERF_ARRANGEMENT_RECORDS),
+            Builtin::View(&MZ_PERF_PEEK_DURATIONS_CORE),
+            Builtin::View(&MZ_PERF_PEEK_DURATIONS_BUCKET),
+            Builtin::View(&MZ_PERF_PEEK_DURATIONS_AGGREGATES),
+            Builtin::View(&MZ_MATERIALIZATION_FRONTIERS),
+            Builtin::View(&MZ_PERF_DEPENDENCY_FRONTIERS),
+        ];
+        builtins.into_iter().map(|b| (b.id(), b)).collect()
```

I think that if you duplicate an id here the map will silently drop the older value so we might want to manually do the insert here and assert that each one doesn't replace anything?

---

i think what might be nice is a separate builtin id map that we insert into that asserts that each item and index id is unique and conforms to the specified ranges? (I am also happy to do that as a follow-on to this)

---

Yep, 100%. I actually wrote that code and then ripped it out. I'm about to land a refactor that maybe deletes the `index_id`s so I didn't want to be fighting against whatever code. I'll file an issue about sanity checking the builtins once all that lands, so we don't forget.